quickstart/101-databricks-cmk-dbfs/main.tf

```diff
@@ -0,0 +1,102 @@
+# Create resource group.
+resource "random_pet" "rg_name" {
+  prefix = var.resource_group_name_prefix
+}
+
+resource "azurerm_resource_group" "rg" {
+  name     = random_pet.rg_name.id
+  location = var.resource_group_location
+}
+
+data "azurerm_client_config" "current" {}
+
+locals {
+  current_user_id = coalesce(var.msi_id, data.azurerm_client_config.current.object_id)
+}
+
+resource "random_pet" "azurerm_databricks_workspace_name" {
+  count  = var.workspace_name == null ? 1 : 0
+  prefix = var.workspace_name_prefix
+}
+
+# Create workspace.
+resource "azurerm_databricks_workspace" "databricks" {
+  name                         = coalesce(var.workspace_name, random_pet.azurerm_databricks_workspace_name[0].id)
+  resource_group_name          = azurerm_resource_group.rg.name
+  location                     = azurerm_resource_group.rg.location
+  sku                          = var.wssku
+  customer_managed_key_enabled = true
+}
+
+# Configure CMK.
+resource "azurerm_databricks_workspace_customer_managed_key" "cmk" {
+  workspace_id     = azurerm_databricks_workspace.databricks.id
+  key_vault_key_id = azurerm_key_vault_key.key[0].id
```

Comment:
Since `azurerm_key_vault_key.key` should not declare `count`, we'd better revert this line back to `key_vault_key_id = azurerm_key_vault_key.key.id`.